core/trie2/triedb/pathdb/disklayer.go

```diff
@@ -0,0 +1,130 @@
+package pathdb
+
+import (
+	"sync"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2/trieutils"
+)
+
+var _ layer = (*diskLayer)(nil)
+
+// Allows access to the underlying database.
+// Nodes are buffered in memory and when the buffer size reaches a certain threshold,
+// the nodes are flushed to the database.
+type diskLayer struct {
+	root    felt.Felt // The corresponding state commitment
+	id      uint64
+	db      *Database
+	cleans  *cleanCache // Clean nodes that are already persisted in the db
+	dirties *buffer     // Dirty nodes for writes later
+	stale   bool
+	lock    sync.RWMutex
+}
+
+func newDiskLayer(root *felt.Felt, id uint64, db *Database, cache *cleanCache, buffer *buffer) *diskLayer {
+	if cache == nil {
+		newCleanCache := newCleanCache(db.config.CleanCacheSize)
+		cache = &newCleanCache
+	}
+	return &diskLayer{
+		root:    *root,
+		id:      id,
+		db:      db,
+		cleans:  cache,
+		dirties: buffer,
+	}
```

Comment:
`stale` is not explicitly initialized here, which means it is false, can we set it here, just so that it's clear